File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/entity/MasterCodeValue.java
package com.nivasafinance.features.master.codemaster.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.nivasafinance.common.audit.AuditableEntity;
import com.nivasafinance.common.base.model.MasterLanguageData;
import io.hypersistence.utils.hibernate.type.json.JsonType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.Type;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "n_master_code_value")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class MasterCodeValue extends AuditableEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "key", nullable = false, length = 100, unique = true)
    private String key;
    
    @Column(name = "code_key", nullable = false, length = 100)
    private String codeKey;
    
    @Type(JsonType.class)
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "value", columnDefinition = "jsonb")
    private MasterLanguageData value;
    
    @Type(JsonType.class)
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "description", columnDefinition = "jsonb")
    private MasterLanguageData description;
    
    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "display_order", nullable = false)
    @Builder.Default
    private Integer displayOrder = 0;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "icons", columnDefinition = "jsonb")
    private IconsData icons;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class IconAsset {
        private String url;
        private Long documentId;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class IconSizeData {
        private IconAsset small;
        private IconAsset medium;
        private IconAsset large;
        private IconAsset xl;
        private IconAsset xxl;
        private IconAsset svg;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class IconsData {
        @JsonProperty("default")
        private IconSizeData defaultIcon;
        private IconSizeData crm;
        private IconSizeData web;
        private IconSizeData app;
    }
}

File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/entity/MasterCodeValue.kt
package com.nivasafinance.features.master.codemaster.entity

import com.nivasafinance.common.audit.AuditableEntity
import com.nivasafinance.common.base.model.MasterLanguageData
import io.hypersistence.utils.hibernate.type.json.JsonType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.annotations.Type
import org.hibernate.type.SqlTypes
import java.util.UUID

@Entity
@Table(name = "master_code_value")
data class MasterCodeValue(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,

    @Column(name = "key", nullable = false, length = 100)
    val key: String,

    @Column(name = "code_key", nullable = false, length = 100)
    val codeKey: String,

    @Type(JsonType::class)
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "value", columnDefinition = "jsonb")
    val value: MasterLanguageData? = null,

    @Type(JsonType::class)
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "description", columnDefinition = "jsonb")
    val description: MasterLanguageData? = null,

    @Column(name = "is_active", nullable = false)
    val isActive: Boolean = true

) : AuditableEntity()
